model/common/src/icon4py/model/common/metrics/metric_fields.py

```diff
@@ -881,205 +774,104 @@ def compute_flat_idx(
 
 
 @field_operator
-def _compute_z_aux2(
+def _compute_downward_extrapolation_distance(
     z_ifc: fa.CellField[wpfloat],
 ) -> fa.EdgeField[wpfloat]:
     extrapol_dist = 5.0
     z_aux1 = maximum(z_ifc(E2C[0]), z_ifc(E2C[1]))
     z_aux2 = z_aux1 - extrapol_dist
-
     return z_aux2
 
 
-@program(grid_type=GridType.UNSTRUCTURED)
-def compute_z_aux2(
-    z_ifc_sliced: fa.CellField[wpfloat],
-    z_aux2: fa.EdgeField[wpfloat],
-    horizontal_start: int32,
-    horizontal_end: int32,
-):
-    _compute_z_aux2(
-        z_ifc=z_ifc_sliced, out=z_aux2, domain={dims.EdgeDim: (horizontal_start, horizontal_end)}
-    )
-
-
 @field_operator
-def _compute_pg_edgeidx_vertidx(
-    c_lin_e: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2CDim], float],
-    z_ifc: fa.CellKField[wpfloat],
+def _compute_pressure_gradient_downward_extrapolation_mask_distance(
+    z_mc: fa.CellKField[wpfloat],
+    c_lin_e: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2CDim], wpfloat],
     z_ifc_sliced: fa.CellField[wpfloat],
     e_owner_mask: fa.EdgeField[bool],
-    flat_idx_max: fa.EdgeField[int32],
-    e_lev: fa.EdgeField[int32],
-    k_lev: fa.KField[int32],
-    pg_edgeidx: fa.EdgeKField[int32],
-    pg_vertidx: fa.EdgeKField[int32],
-) -> tuple[fa.EdgeKField[int32], fa.EdgeKField[int32]]:
-    z_aux2 = _compute_z_aux2(z_ifc_sliced)
+    flat_idx_max: fa.EdgeField[gtx.int32],
+    e_lev: fa.EdgeField[gtx.int32],
+    k_lev: fa.KField[gtx.int32],
+    horizontal_start_distance: int32,
+    horizontal_end_distance: int32,
+) -> tuple[fa.EdgeKField[bool], fa.EdgeKField[wpfloat]]:
+    """
+    Compute an edge mask and extrapolation distance for grid points requiring downward extrapolation of the pressure gradient.
+
+    See pg_edgeidx and pg_exdist in mo_vertical_grid.f90
+
+    Args:
+        z_mc:
+        c_lin_e:
+        z_ifc_sliced:
+        e_owner_mask:
+        flat_idx_max:
+        e_lev:
+        k_lev:
+        horizontal_start_distance:
+        horizontal_end_distance:
```

Comment:
```suggestion
        z_mc: local field
        c_lin_e: interpolation field
        z_ifc_sliced: z_ifc sliced at nlev
        e_owner_mask: edge owner mask
        flat_idx_max: max value for flat_idx
        e_lev: edge levels field
        k_lev: vertical levels field
        horizontal_start_distance: horizontal upper bound
        horizontal_end_distance: horizontal lower bound
```

Reply:
but I would move this docstring to the program definition

Reply:
I generally prefer to have them on the `field_operator`, as we might get rid of the program...